pkg/controller/connectionsecret/listsecrets.go

```diff
@@ -0,0 +1,56 @@
+package connectionsecret
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+)
+
+// ListByClusterName returns all secrets in the specified namespace that have labels for 'projectID' and 'clusterName'
+func ListByClusterName(k8sClient client.Client, namespace, projectID, clusterName string) ([]corev1.Secret, error) {
+	return list(k8sClient, namespace, projectID, clusterName, "")
+}
+
+// ListByUserName returns all secrets in the specified namespace that have label for 'projectID' and data for 'userName'
+func ListByUserName(k8sClient client.Client, namespace, projectID, userName string) ([]corev1.Secret, error) {
+	return list(k8sClient, namespace, projectID, "", userName)
+}
+
+func list(k8sClient client.Client, namespace, projectID, clusterName, dbUserName string) ([]corev1.Secret, error) {
+	secrets := corev1.SecretList{}
+	var result []corev1.Secret
+	if err := k8sClient.List(context.Background(), &secrets, client.InNamespace(namespace)); err != nil {
+		return nil, err
+	}
+
+	for _, s := range secrets.Items {
+		if value, ok := s.Labels[ProjectLabelKey]; !ok || value != projectID {
+			continue
+		}
+		if _, ok := s.Labels[ClusterLabelKey]; !ok {
+			continue
+		}
+		if clusterName != "" && s.Labels[ClusterLabelKey] == kube.NormalizeLabelValue(clusterName) {
+			result = append(result, s)
+		}
+		if dbUserName != "" {
+			secretData := make(map[string]string)
+			for k, v := range s.Data {
```

Comment:
[q] why do we need to create a copy of the map?

Reply:
`s.Data` is a `map[string][]byte` - so we need to read each value and transform to string

Reply:
but I see what you mean - we can just read the value and cast to string. Will do